File: Components/DevExpressVCL/Demos/ExpressOrgChart/CBuilder/Options.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "Options.h"
//---------------------------------------------------------------------------
#pragma link "cxControls"
#pragma link "cxGraphics"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxLookAndFeels"
#pragma resource "*.dfm"
TOptionsForm *OptionsForm;
//---------------------------------------------------------------------------
__fastcall TOptionsForm::TOptionsForm(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TOptionsForm::FormActivate(TObject *Sender)
{
  if (MainForm->PC->ActivePage == MainForm->tsOrgChart) {
    Caption = "dxOrgChart Options";
    cbLeft->Checked     = MainForm->Tree->EditMode.Contains(emLeft);
    cbCenter->Checked   = MainForm->Tree->EditMode.Contains(emCenter);
    cbRight->Checked    = MainForm->Tree->EditMode.Contains(emRight);
    cbVCenter->Checked  = MainForm->Tree->EditMode.Contains(emVCenter);
    cbWrap->Checked     = MainForm->Tree->EditMode.Contains(emWrap);
    cbUpper->Checked    = MainForm->Tree->EditMode.Contains(emUpper);
    cbLower->Checked    = MainForm->Tree->EditMode.Contains(emLower);
    cbGrow->Checked     = MainForm->Tree->EditMode.Contains(emGrow);

    cbSelect->Checked   = MainForm->Tree->Options.Contains(ocSelect);
    cbFocus->Checked    = MainForm->Tree->Options.Contains(ocFocus);
    cbButtons->Checked  = MainForm->Tree->Options.Contains(ocButtons);
    cbEdit->Checked     = MainForm->Tree->Options.Contains(ocEdit);
    cbCanDrag->Checked  = MainForm->Tree->Options.Contains(ocCanDrag);
    cbShowDrag->Checked = MainForm->Tree->Options.Contains(ocShowDrag);
    cbInsDel->Checked   = MainForm->Tree->Options.Contains(ocInsDel);

	seX->Text = IntToStr(MainForm->Tree->IndentX);
	seY->Text = IntToStr(MainForm->Tree->IndentY);
    seLineWidth->Text = IntToStr(MainForm->Tree->LineWidth);
    cbShowImages->Checked = ! (MainForm->Tree->Images == NULL);
  };

  if (MainForm->PC->ActivePage == MainForm->tsDBOrgChart) {
    Caption = "dxDBOrgChart Options";
    cbLeft->Checked     = MainForm->DBTree->EditMode.Contains(emLeft);
    cbCenter->Checked   = MainForm->DBTree->EditMode.Contains(emCenter);
    cbRight->Checked    = MainForm->DBTree->EditMode.Contains(emRight);
    cbVCenter->Checked  = MainForm->DBTree->EditMode.Contains(emVCenter);
    cbWrap->Checked     = MainForm->DBTree->EditMode.Contains(emWrap);
    cbUpper->Checked    = MainForm->DBTree->EditMode.Contains(emUpper);
    cbLower->Checked    = MainForm->DBTree->EditMode.Contains(emLower);
    cbGrow->Checked     = MainForm->DBTree->EditMode.Contains(emGrow);

    cbSelect->Checked   = MainForm->DBTree->Options.Contains(ocSelect);
    cbFocus->Checked    = MainForm->DBTree->Options.Contains(ocFocus);
    cbButtons->Checked  = MainForm->DBTree->Options.Contains(ocButtons);
    cbEdit->Checked     = MainForm->DBTree->Options.Contains(ocEdit);
    cbCanDrag->Checked  = MainForm->DBTree->Options.Contains(ocCanDrag);
    cbShowDrag->Checked = MainForm->DBTree->Options.Contains(ocShowDrag);
    cbInsDel->Checked   = MainForm->DBTree->Options.Contains(ocInsDel);

	seX->Text = IntToStr(MainForm->DBTree->IndentX);
	seY->Text = IntToStr(MainForm->DBTree->IndentY);
    seLineWidth->Text = IntToStr(MainForm->DBTree->LineWidth);
    cbShowImages->Checked = ! (MainForm->DBTree->Images == NULL);
  };
}
//---------------------------------------------------------------------------
void __fastcall TOptionsForm::BitBtn2Click(TObject *Sender)
{
  if (MainForm->PC->ActivePage == MainForm->tsOrgChart) {
    if (cbLeft->Checked) MainForm->Tree->EditMode << emLeft;
    else MainForm->Tree->EditMode >> emLeft;
    if (cbCenter->Checked) MainForm->Tree->EditMode << emCenter;
    else MainForm->Tree->EditMode >> emCenter;
    if (cbRight->Checked) MainForm->Tree->EditMode << emRight;
    else MainForm->Tree->EditMode >> emRight;
    if (cbVCenter->Checked) MainForm->Tree->EditMode << emVCenter;
    else MainForm->Tree->EditMode >> emVCenter;
    if (cbWrap->Checked) MainForm->Tree->EditMode << emWrap;
    else MainForm->Tree->EditMode >> emWrap;
    if (cbUpper->Checked) MainForm->Tree->EditMode << emUpper;
    else MainForm->Tree->EditMode >> emUpper;
    if (cbLower->Checked) MainForm->Tree->EditMode << emLower;
    else MainForm->Tree->EditMode >> emLower;
    if (cbGrow->Checked) MainForm->Tree->EditMode << emGrow;
    else MainForm->Tree->EditMode >> emGrow;

    if (cbSelect->Checked) MainForm->Tree->Options << ocSelect;
    else MainForm->Tree->Options >> ocSelect;
    if (cbFocus->Checked) MainForm->Tree->Options << ocFocus;
    else MainForm->Tree->Options >> ocFocus;
    if (cbButtons->Checked) MainForm->Tree->Options << ocButtons;
    else MainForm->Tree->Options >> ocButtons;
    if (cbEdit->Checked) MainForm->Tree->Options << ocEdit;
    else MainForm->Tree->Options >> ocEdit;
    if (cbCanDrag->Checked) MainForm->Tree->Options << ocCanDrag;
    else MainForm->Tree->Options >> ocCanDrag;
    if (cbShowDrag->Checked) MainForm->Tree->Options << ocShowDrag;
    else MainForm->Tree->Options >> ocShowDrag;
    if (cbInsDel->Checked) MainForm->Tree->Options << ocInsDel;
    else MainForm->Tree->Options >> ocInsDel;
    if (cbShowImages->Checked) MainForm->Tree->Images = MainForm->ilTree;
    else MainForm->Tree->Images = NULL;

	MainForm->Tree->IndentX = StrToInt(seX->Text);
	MainForm->Tree->IndentY = StrToInt(seY->Text);
    MainForm->Tree->LineWidth = StrToInt(seLineWidth->Text);
    MainForm->Tree->Refresh();
  };

  if (MainForm->PC->ActivePage == MainForm->tsDBOrgChart) {
    if (cbLeft->Checked) MainForm->DBTree->EditMode << emLeft;
    else MainForm->DBTree->EditMode >> emLeft;
    if (cbCenter->Checked) MainForm->DBTree->EditMode << emCenter;
    else MainForm->DBTree->EditMode >> emCenter;
    if (cbRight->Checked) MainForm->DBTree->EditMode << emRight;
    else MainForm->DBTree->EditMode >> emRight;
    if (cbVCenter->Checked) MainForm->DBTree->EditMode << emVCenter;
    else MainForm->DBTree->EditMode >> emVCenter;
    if (cbWrap->Checked) MainForm->DBTree->EditMode << emWrap;
    else MainForm->DBTree->EditMode >> emWrap;
    if (cbUpper->Checked) MainForm->DBTree->EditMode << emUpper;
    else MainForm->DBTree->EditMode >> emUpper;
    if (cbLower->Checked) MainForm->DBTree->EditMode << emLower;
    else MainForm->DBTree->EditMode >> emLower;
    if (cbGrow->Checked) MainForm->DBTree->EditMode << emGrow;
    else MainForm->DBTree->EditMode >> emGrow;

    if (cbSelect->Checked) MainForm->DBTree->Options << ocSelect;
    else MainForm->DBTree->Options >> ocSelect;
    if (cbFocus->Checked) MainForm->DBTree->Options << ocFocus;
    else MainForm->DBTree->Options >> ocFocus;
    if (cbButtons->Checked) MainForm->DBTree->Options << ocButtons;
    else MainForm->DBTree->Options >> ocButtons;
    if (cbEdit->Checked) MainForm->DBTree->Options << ocEdit;
    else MainForm->DBTree->Options >> ocEdit;
    if (cbCanDrag->Checked) MainForm->DBTree->Options << ocCanDrag;
    else MainForm->DBTree->Options >> ocCanDrag;
    if (cbShowDrag->Checked) MainForm->DBTree->Options << ocShowDrag;
    else MainForm->DBTree->Options >> ocShowDrag;
    if (cbInsDel->Checked) MainForm->DBTree->Options << ocInsDel;
    else MainForm->DBTree->Options >> ocInsDel;
    if (cbShowImages->Checked) MainForm->DBTree->Images = MainForm->ilTree;
    else MainForm->DBTree->Images = NULL;

	MainForm->DBTree->IndentX = StrToInt(seX->Text);
	MainForm->DBTree->IndentY = StrToInt(seY->Text);
    MainForm->DBTree->LineWidth = StrToInt(seLineWidth->Text);
    MainForm->DBTree->Refresh();
  };

  Close();
}
//---------------------------------------------------------------------------

File: Components/DevExpressVCL/Demos/ExpressOrgChart/Delphi/Options.pas
unit Options;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  StdCtrls, Spin, Buttons, ExtCtrls, dxorgchr, cxGraphics, cxLookAndFeels,
  cxLookAndFeelPainters, Menus, cxControls;

type
  TOptionsForm = class(TForm)
    Bevel1: TBevel;
    cbButtons: TCheckBox;
    cbCanDrag: TCheckBox;
    cbCenter: TCheckBox;
    cbEdit: TCheckBox;
    cbFocus: TCheckBox;
    cbGrow: TCheckBox;
    cbInsDel: TCheckBox;
    cbLeft: TCheckBox;
    cbLower: TCheckBox;
    cbRight: TCheckBox;
    cbSelect: TCheckBox;
    cbShowDrag: TCheckBox;
    cbShowImages: TCheckBox;
    cbUpper: TCheckBox;
    cbVCenter: TCheckBox;
    cbWrap: TCheckBox;
    GroupBox1: TGroupBox;
    Label1: TLabel;
    Label2: TLabel;
    Label3: TLabel;
    seLineWidth: TSpinEdit;
    seX: TSpinEdit;
    seY: TSpinEdit;
    BitBtn1: TButton;
    BitBtn2: TButton;
    procedure BitBtn2Click(Sender: TObject);
    procedure FormActivate(Sender: TObject);
  end;

var
  OptionsForm: TOptionsForm;

implementation

uses
  main;

{$R *.DFM}

type
  TdxCustomOrgChartAccess = class(TdxCustomOrgChart);

{ TOptionsForm }

procedure TOptionsForm.FormActivate(Sender: TObject);
begin
  with TdxCustomOrgChartAccess(MainForm.ActiveOrgChart) do
  begin
    cbLeft.Checked := emLeft in EditMode;
    cbCenter.Checked := emCenter in EditMode;
    cbRight.Checked := emRight in EditMode;
    cbVCenter.Checked := emVCenter in EditMode;
    cbWrap.Checked := emWrap in EditMode;
    cbUpper.Checked := emUpper in EditMode;
    cbLower.Checked := emLower in EditMode;
    cbGrow.Checked := emGrow in EditMode;

    cbSelect.Checked := ocSelect in Options;
    cbFocus.Checked := ocFocus in Options;
    cbButtons.Checked := ocButtons in Options;
    cbEdit.Checked := ocEdit in Options;
    cbCanDrag.Checked := ocCanDrag in Options;
    cbShowDrag.Checked := ocshowDrag in Options;
    cbInsDel.Checked := ocInsDel in Options;

    seX.Value := IndentX;
    seY.Value := IndentY;
    seLineWidth.Value := LineWidth;
    cbShowImages.Checked := not(Images = nil);
  end;
end;

procedure TOptionsForm.BitBtn2Click(Sender: TObject);
begin
  with TdxCustomOrgChartAccess(MainForm.ActiveOrgChart) do
  begin
    if cbLeft.Checked then EditMode := EditMode + [emLeft] else EditMode := EditMode - [emLeft];
    if cbCenter.Checked then EditMode := EditMode + [emCenter] else EditMode := EditMode - [emCenter];
    if cbRight.Checked then EditMode := EditMode + [emRight] else EditMode := EditMode - [emRight];
    if cbVCenter.Checked then EditMode := EditMode + [emVCenter] else EditMode := EditMode - [emVCenter];
    if cbWrap.Checked then EditMode := EditMode + [emWrap] else EditMode := EditMode - [emWrap];
    if cbUpper.Checked then EditMode := EditMode + [emUpper] else EditMode := EditMode - [emUpper];
    if cbLower.Checked then EditMode := EditMode + [emLower] else EditMode := EditMode - [emLower];
    if cbGrow.Checked then EditMode := EditMode + [emGrow] else EditMode := EditMode - [emGrow];

    if cbSelect.Checked then Options := Options + [ocSelect] else Options := Options - [ocSelect];
    if cbFocus.Checked then Options := Options + [ocFocus] else Options := Options - [ocFocus];
    if cbButtons.Checked then Options := Options + [ocButtons] else Options := Options - [ocButtons];
    if cbEdit.Checked then Options := Options + [ocEdit] else Options := Options - [ocEdit];
    if cbCanDrag.Checked then Options := Options + [ocCanDrag] else Options := Options - [ocCanDrag];
    if cbShowDrag.Checked then Options := Options + [ocShowDrag] else Options := Options - [ocShowDrag];
    if cbInsDel.Checked then Options := Options + [ocInsDel] else Options := Options - [ocInsDel];
    if CbShowImages.Checked then Images := MainForm.ilTree else Images := nil;

    IndentX := seX.Value;
    IndentY := seY.Value;
    LineWidth := seLineWidth.Value;
  end;
end;

end.
